src/nyx/features/glrlm.cpp

```diff
@@ -518,24 +532,27 @@ void GLRLMFeature::calc_SRE (AngledFtrs& af)
 		return;
 	}
 
+
 	for (int ai = 0; ai < 4; ai++)
 	{
 		// Get ahold of the requested angle's matrix and its related N parameters 
 		int Ng = angles_Ng[ai],
 			Nr = angles_Nr[ai];
 		const SimpleMatrix<int>& P = angles_P[ai];
 
-		// Calculate
-		double f = 0.0;
-		for (int i = 1; i <= Ng; i++)
-		{
-			for (int j = 1; j <= Nr; j++)
-			{
-				f += P.matlab(i, j) / (j * j);
+		double f = 0.;
+		std::vector<double> rj(Nr+1, 0.);
+		for (int i = 1; i <= Ng; ++i) {
+			for (int j = 1; j <= Nr; ++j) {
+				rj[j] += P.matlab(i, j);
 			}
 		}
 
-		double retval = f / double(Nr);
+		for (int j = 1; j <= Nr; ++j) {
+			f +=  rj[j] / (j * j);
+		}
+
+		double retval = f / double(sum_p[ai]);

```

Comment:
No need to cast, 'sum_p' is a container of doubles already.
(Here and below.)

Reply:
Removed